src/apps/uniswap/positions.ts

```diff
@@ -1,19 +1,49 @@
 import { Address } from 'viem'
+import { getClient } from '../../runtime/client'
+import { getTokenId } from '../../runtime/getTokenId'
+import { NetworkId } from '../../types/networkId'
 import { toDecimalNumber } from '../../types/numbers'
 import { PositionsHook } from '../../types/positions'
 import { userPositionsAbi } from './abis/user-positions'
-import { getClient } from '../../runtime/client'
-import { NetworkId } from '../../types/networkId'
-import { getTokenId } from '../../runtime/getTokenId'
 
-// Standard Uniswap v3 addresses on CELO
-const UNISWAP_V3_FACTORY_ADDRESS = '0xAfE208a311B21f13EF87E33A90049fC17A7acDEc'
-const UNISWAP_V3_NFT_POSITIONS_MANAGER_ADDRESS =
-  '0x3d79EdAaBC0EaB6F08ED885C05Fc0B014290D95A'
-// Custom read-only contract. Code:
-// https://github.com/celo-tracker/celo-tracker-contracts/blob/main/contracts/multicall/UniV3UserPositionsMulticall.sol
-const USER_POSITIONS_MULTICALL_ADDRESS =
-  '0x0588Cc1eD79e3c754F4180E78554691E82c2dEdB'
+const UNI_V3_ADDRESSES: {
+  [networkId: string]:
+    | {
+        factory: Address
+        nftPositions: Address
+        // Custom read-only contract. Code:
+        // https://github.com/celo-tracker/celo-tracker-contracts/blob/main/contracts/multicall/UniV3UserPositionsMulticall.sol
+        userPositionsMulticall: Address
```

Comment:
oh this reminds me, viem just added support for deployless calls,
nothing to do change in this PR, but what do you think of trying it in the future to see if it would alleviate the need to actually deploy the multicall contract?

https://x.com/_jxom/status/1802530297919193507

Reply:
This would be great, specially for Ethereum where the cost of deploying a contract can be really high!!